src/core_ghidra.cpp

```diff
@@ -132,6 +133,87 @@ static void ApplyPrintCConfig(RConfig *cfg, PrintC *print_c)
 	print_c->setMaxLineSize(cfg_var_linelen.GetInt(cfg));
 }
 
+RAnnotatedCode* r2ghidra_decompile_annotated_code(RCore *core, ut64 addr){
+	DecompilerLock lock;
+	RAnnotatedCode *code = nullptr;
+#ifndef DEBUG_EXCEPTIONS
```

Comment:
I don't think it's necessary to keep old hacky debug code. Also the same can be easily achieved by setting breakpoint on exceptions or in the catch statement.

Reply:
@karliss  Can you please look into the code and suggest parts you think I should separate? I had asked for suggestions to refactor for avoiding code duplication in the description of this PR.  I am not really sure about what all should be changed.